### PR TITLE
client: Run execution in batches when not near head

### PR DESCRIPTION
**Pull request by g11tech**
Run execution in batches when not near head

### CODE REVIEW

**Review by acolytec3 — COMMENTED** [review]
LGTM!

**Review by acolytec3 — APPROVED** [review]

**Review by acolytec3 — APPROVED** [review]
still lgtm

File: packages/client/lib/sync/fetcher/blockfetcher.ts [hunk 1]
```diff
@@ -48,7 +48,7 @@ export class BlockFetcher extends BlockFetcherBase<Block[], Block> {
     }
     const headers = headersResult[1]
     const bodiesResult = await peer!.eth!.getBlockBodies({ hashes: headers.map((h) => h.hash()) })
-    if (isFalsy(bodiesResult) || bodiesResult[1].length === 0) {
+    if (isFalsy(bodiesResult) || isFalsy(bodiesResult[1]) || bodiesResult[1].length === 0) {
```
Comment (holgerd77) [inline packages/client/lib/sync/fetcher/blockfetcher.ts:51]: Can we try not to use respectively even introduce these `isFalsy` (and friends 🙂) methods but directly go to the explicit version?

This was really intended to just be a placeholder until we do better.

Comment (holgerd77) [reply packages/client/lib/sync/fetcher/blockfetcher.ts:51]: (or is the whole concept not working/practicable and we would need to revise the linting rule update at this point?)

**Pull request merged**